src/main.ts

```diff
@@ -345,41 +358,73 @@ export default class ReadwisePlugin extends Plugin {
     window.clearInterval(this.scheduleInterval);
     this.scheduleInterval = null;
     if (!milliseconds) {
-      // we got manual option
+      // user set frequency to manual
       return;
     }
-    this.scheduleInterval = window.setInterval(() => this.requestArchive(null, null, true), milliseconds);
+    this.scheduleInterval = window.setInterval(() => this.refreshBookExport(undefined, true), milliseconds);
     this.registerInterval(this.scheduleInterval);
   }
 
-  refreshBookExport(bookIds?: Array<string>) {
-    bookIds = bookIds || this.settings.booksToRefresh;
-    if (!bookIds.length || !this.settings.refreshBooks) {
+  /** Initiates a sync, or queues refresh for later if queue fails.
+   * ALL syncing goes through this function. */
+  async refreshBookExport(
+    /** optional list of specific book IDs to sync */
+    bookIds?: Array<string>,
+
+    /** if true, was not initiated by user */
+    auto?: boolean,
+  ) {
+    if (!this.settings.token) return;
+
+    const targetBookIds = bookIds || this.settings.booksToRefresh;
+
+    // add potentially-missing books to booksToRefresh (TODO - prob a lil inefficient? 🤷)
+    const knownFilesPaths = Object.keys(this.settings.booksIDsMap);
+    if (this.settings.refreshBooks) {
+      for (const knownFilePath of knownFilesPaths) {
+        const file = this.app.vault.getAbstractFileByPath(knownFilePath);
+        if (!file) {
+          const bookId = this.settings.booksIDsMap[knownFilePath];
+          targetBookIds.push(bookId);
+        }
+      }
+    }
+
+    const hasNeverSynced = !Object.keys(this.settings.booksIDsMap).length;
+    if (hasNeverSynced) {
+      this.notice("Preparing initial Readwise sync...", true);
+      await this.startSync();
+      return;
+    }
+
+    if (!targetBookIds.length) {
+      console.log('Readwise Official plugin: no targetBookIds, triggering check for other updates...');
+      await this.requestArchive(null, null, auto);

```

Comment:
hmm... this won't set the `isSyncing` setting that happens via `startSync` function... did u miss this or for some reason it's not a problem?

Reply:
Looks like an oversight, but I think I have a solution here: https://github.com/readwiseio/obsidian-readwise/pull/67#discussion_r1739794857

Reply:
I did the proposed solution in the previous comment :arrow_heading_up: so the `isSyncing` should now getting set here :cowboy_hat_face: